python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -819,16 +827,50 @@ def GetCurrentImage(self, request, _context):
             )
         )
 
-    def GetCurrentRuns(self, request, context) -> api_pb2.GetCurrentRunsReply:  # type: ignore
+    def GetCurrentRuns(self, request, context) -> api_pb2.GetCurrentRunsReply:
         with self._execution_lock:
-            return api_pb2.GetCurrentRunsReply(  # type: ignore
+            return api_pb2.GetCurrentRunsReply(
                 serialized_current_runs=serialize_value(
                     GetCurrentRunsResult(
                         current_runs=list(self._executions.keys()), serializable_error_info=None
                     )
                 )
             )
 
+    def _resource_verification(
+        self, resource_verification_request: ResourceVerificationRequest
+    ) -> ResourceVerificationResult:
+        definition = self._get_repo_for_origin(resource_verification_request.repository_origin)
+        return launch_resource_verification(
+            repo_def=definition,
+            instance_ref=self._instance_ref,
+            resource_name=resource_verification_request.resource_name,
+        )
+
+    def UserCodeExecution(self, request: Any, _context) -> api_pb2.UserCodeExecutionResult:
+        user_code_execution_request = deserialize_value(
+            request.serialized_user_code_execution_request, UserCodeExecutionRequest
+        )
+        if (
+            user_code_execution_request.execution_type
+            == UserCodeExecutionType.RESOURCE_VERIFICATION
+        ):
+            result = self._resource_verification(
+                user_code_execution_request.unpack_as(ResourceVerificationRequest)
+            )
+        else:
+            raise NotImplementedError(
+                "Unsupported user code execution type:"
+                f" {user_code_execution_request.execution_type}"
+            )
+        return api_pb2.UserCodeExecutionResult(
+            serialized_user_code_execution_result=serialize_value(
+                UserCodeExecutionResult(
+                    execution_type=user_code_execution_request.execution_type, data=result
+                )
+            )
+        )
```

Comment:
some ideas to consider for this set-up 
 * Just unpack the request in a try/catch and if it fails send raise "unvailable on version ..." exception chained with the deser exception. This would handle both top level missing request types and any problems on inner contents (though those should be unlikely). 
 * Make these serdes tuples implement an interface like `execute(self, ...)` and just call that here. Need to figure out what exactly to pass in - maybe `LoadedRepostiories` instead of the whole server. That part can change over time so not super critical to get right. 
 
 Its possible we can get nice behavior by actually trying to the appropriate grpc / protobuf primitives but we have talked about moving away from GRPC at somepoint which remains quite feasible given our low investment in their abstractions.